tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/messaging/artemis/LocalTransactionTest.java

```diff
@@ -37,47 +35,39 @@
  */
 @Test(groups = {"artemis-test"})
 public class LocalTransactionTest {
-    private CompileResult producerResult, consumerResult;
+    private CompileResult erringTransactResult;
+    private CompileResult transactConsumerResult;
+    private CompileResult transactSimpleConsumerResult;
+    private static final String INVALID_MESSAGE = "Invalid message received";
 
     @BeforeClass
     public void setup() throws URISyntaxException {
         TestUtils.prepareBalo(this);
         Path sourcePath = Paths.get("src", "test", "resources", "messaging", "artemis", "src");
-        producerResult = BCompileUtil.compile(sourcePath.resolve("producers").resolve("transaction_producer.bal")
-                                                      .toAbsolutePath().toString());
-        consumerResult = BCompileUtil.compile(sourcePath.resolve("consumers").resolve("transaction_consumer.bal")
-                                                      .toAbsolutePath().toString());
+        erringTransactResult = BCompileUtil.compile(sourcePath.resolve("transaction").resolve("erring_transaction.bal")
+                .toAbsolutePath().toString());
+        transactConsumerResult = BCompileUtil.compile(sourcePath.resolve("transaction").resolve("consumer.bal")
+                .toAbsolutePath().toString());
+        transactSimpleConsumerResult = BCompileUtil.compile(sourcePath.resolve("transaction")
+                .resolve("simple_consumer.bal").toAbsolutePath().toString());
     }
 
     @Test(description = "Tests the sending of a string message to a queue")
     public void testSimpleSend() {
-        //Invoking this function is needed to make sure the queue is created before the sending
-        BValue[] consumerVal = BRunUtil.invoke(consumerResult, "createSimpleConsumer");
-        BRunUtil.invoke(producerResult, "testSimpleTransactionSend");
-        String returnVal = BRunUtil.invoke(consumerResult, "transactionSimpleConsumerReceive", consumerVal)[0]
+        String returnVal = BRunUtil.invoke(transactSimpleConsumerResult, "testSimpleTransaction")[0]
                 .stringValue();
-        Assert.assertEquals(returnVal, "Example Example ", "Invalid message received");
+        Assert.assertEquals(returnVal, "Example Example ", INVALID_MESSAGE);
```

Comment:
Shall we move the invoking and asserting code to a different method to avoid duplication?

Reply:
Resolved